### PR TITLE
Add PCA decomposition algorithm support

### LINKED ISSUE #513: Implement PCA decomposition algorithm
**Is your feature request related to a problem? Please describe.**

We can implement PCA decomposition as a part of `mars.learn`. 

**Describe the solution you'd like**

We can replicate the implementation in sklearn: https://github.com/scikit-learn/scikit-learn/blob/ec6d0bb2481dd756b7e992659ea0e7f864873916/sklearn/decomposition/pca.py#L104, but cautiously take care of tensor instead of ndarray.

As for the `svd_solver`, we have not parallelized `svds` yet, so we can provide solver including 'auto', 'full', 'randomized'. 

`randomized_svd` is introduced in #388 .

### PR DESCRIPTION
**Pull request by qinxuye**
## What do these changes do?

This PR adds the PCA decomposition algorithm.

TODO:

- [x] Wait for merging of #628 .
- [x] Add more unittests.

## Related issue number

Resolves #513 .

### CODE REVIEW

File: mars/fuse.pyx [hunk 1]
```diff
@@ -96,30 +96,27 @@ class Fusion(object):
...
                 pre = get_node(pre)
                 if pre in nodes_set:
                     self._graph.add_node(pre)
                 if pre in self._graph:
                     self._graph.add_edge(pre, cur_node)
                 if pre in nodes_set:
                     q.appendleft(pre)
-                    observed.add(pre)
-        if len(observed) != len(nodes_set):
-            raise InvalidComposedNodeError("Invalid composed node data")
```
Comment (wjsi) [inline mars/fuse.pyx]: The error definition can be removed when it is not needed.

File: mars/tensor/datasource/array.py [hunk 2]
```diff
@@ -151,9 +152,18 @@ def tensor(data, dtype=None, order='K', chunk_size=None, gpu=None, sparse=False)
         return scalar(data, dtype=dtype)
     elif issparse(data):
         return _from_spmatrix(data, dtype=dtype, chunk_size=chunk_size, gpu=gpu)
+    elif hasattr(data, '__mars_tensor__'):
+        data = data.__mars_tensor__(dtype=dtype)
+        return data.astype(data.dtype, order=order, copy=False)
```
Comment (wjsi) [inline mars/tensor/datasource/array.py]: Now that we already do ``data.astype`` in ``__mars_tensor__``, we may pass order as an argument as well.

File: mars/fuse.pyx (hunk 1 above)
Comment (qinxuye) [reply mars/fuse.pyx]: Removed already.

File: mars/tensor/datasource/array.py (hunk 2 above)
Comment (qinxuye) [reply mars/tensor/datasource/array.py]: Done.

**Pull request merged**